tests/unit/test_config_manager.py

```diff
@@ -608,3 +608,27 @@ def test_get_none_run_id(module_under_test):
         SAMPLE_CONFIG, MockIbisClient(), MockIbisClient(), verbose=False
     )
     assert config_manager.run_id is None
+
+
+@mock.patch(
+    "data_validation.config_manager.ConfigManager.get_source_ibis_calculated_table",
+    new=lambda x: MockIbisTable(),
+)
+def test_build_comp_fields(module_under_test):
+    config_manager = module_under_test.ConfigManager(
+        SAMPLE_CONFIG, MockIbisClient(), MockIbisClient(), verbose=False
+    )
+
+    # With exclude_columns=False
+    comparison_fields = config_manager.build_comp_fields(
+        ["a", "c", "e"],
+        False,
+    )
+    assert comparison_fields == {"a": "a", "c": "c"}
```

Comment:
To confirm my understanding: column "e" isn't in the expected dict because  it only exists in the source table and not in the target? If not, not sure I understood here.

Reply:
Column "e" is not in either table and therefore is not expecting in the final list, I've added a comment to that effect.